src/System/Random/Stateful.hs

```diff
@@ -236,25 +238,56 @@ withMutableGen fg action = do
 
 -- | Same as 'withMutableGen', but only returns the generated value.
 --
+-- ====__Examples__
+--
+-- >>> import System.Random.Stateful
+-- >>> let pureGen = mkStdGen 137
+-- >>> withMutableGen_ (IOGen pureGen) (uniformRM (1 :: Int, 6 :: Int))
+-- 4
+--
 -- @since 1.2.0
 withMutableGen_ :: FrozenGen f m => f -> (MutableGen f m -> m a) -> m a
 withMutableGen_ fg action = fst <$> withMutableGen fg action
 
 
 -- | Generates a list of pseudo-random values.
 --
+-- ====__Examples__
+--
+-- >>> import System.Random.Stateful
+-- >>> let pureGen = mkStdGen 137
+-- >>> g <- newIOGenM pureGen
+-- >>> uniformListM g 10 :: IO [Bool]
+-- [True,True,True,True,False,True,True,False,False,False]
+--
 -- @since 1.2.0
 uniformListM :: (StatefulGen g m, Uniform a) => g -> Int -> m [a]
 uniformListM gen n = replicateM n (uniformM gen)
 
 -- | Generates a pseudo-random value using monadic interface and `Random` instance.
 --
+-- ====__Examples__
+--
+-- >>> import System.Random.Stateful
```

Comment:
I don't really like this replacement since it makes use of an implicit monad in ghci. The purpose of examples is usually for people (typically beginners) who cannot follow the types and I think it is very confusing to make use of such features. That might just be me though, so I would be interested what other people think. If using curlies is considered bad I think:
```haskell
 (newIOGenM pureGen >>= randomM g) :: IO Double
```
is preferable.

Reply:
ghci is the first thing people learn, so I don't really see a problem with current example.

I won't object switching to `>>=`, but I think it is much harder for novice users to understand bind when compared to regular `do` notation.

I am definitely against curly braces. This is such rarely used syntax that it will only confuse people.

Reply:
That is fair, if you think people will easily figure out how to translate it to a file then I am all for it.

Reply:
Translation is literally one to one:
```haskell
>>> let pureGen = mkStdGen 137
>>> g <- newIOGenM pureGen
>>> randomM g :: IO Double
```
to
```haskell
... = do
  let pureGen = mkStdGen 137
  g <- newIOGenM pureGen
  randomM g :: IO Double
```

And if someone has a problem going from ghci to a monadic function they have bigger problems and probably should stick to pure API and manual generator passing with tuples ;)